ferrilata-devilline/Controllers/ApiAdminController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using ferrilata_devilline.Services.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace ferrilata_devilline.Controllers
+{
+    [ApiController]
+    public class ApiAdminController : Controller
+    {
+        [HttpPost("api/admin/add")]
+        public IActionResult AddAdmin([FromBody]JObject data)
+        {
+            var header = Request.Headers;
+            string authorization = header.ContainsKey("Authorization") ? (string)header["Authorization"] : null;
+
+            if (authorization != null && authorization != "")
+            {
+                if (data == null || data.HasMissingFieldsAsAdmin() || data.HasNullValuesAsAdmin()) 
+                {
+                    var missingFieldResult = new ObjectResult(new { error = "Please provide all fields" });
+                    missingFieldResult.StatusCode = 400; 
+                    return missingFieldResult;
```

Comment:
This could be more simple like this: `return BadRequest(new { error = "Please provide all fields" });`